### PR TITLE
Handle case where mutliple disabilities could have same diagnostic code.

### PR DESCRIPTION
**Pull request by dfitchett**
## Summary
- This work is behind a feature toggle (flipper): NO
- This PR adjusts the logic to handle multiple disabilities with the same diagnostic code when assigning the `maximum_rating` to disabilities returned in the rated disabilities response used to populate information on the rated disabilities page of the Form 526EZ. 
- I am part of the Claims Fast-Tracking crew's Employee Experience team. We work closely with the Disability Experience team which owns this code.

## Related issue(s)
- discovered during work on another ticket related to displaying the maximum rating: https://github.com/department-of-veterans-affairs/abd-vro/issues/2722 
- ticket to fix: https://github.com/department-of-veterans-affairs/abd-vro/issues/2849

## Testing done
- New code is covered by new unit tests for multiple disabilities with same diagnostic code
- Old unit tests still passing
- If rated disabilities included multiple disabilities with the same diagnostic code (e.g., tinnitus of left/right ear), the maximum rating notice would only be shown on the first. 
- Running locally:
  - enable toggle `disability_526_maximum_rating_api_all_conditions` so below users conditions are prefilled with max rating
  - log in with vets.gov.user+1@gmail.com
  - start new [526 form](http://localhost:3001/disability/file-disability-claim-form-21-526ez/introduction) to allow prefill of maximum rating data
  - progress to rated disabilities page to see both conditions with maximum rating notice

## Screenshots
| Before | After | 
| --- | --- |
| Screenshot 2024-04-10 at 9 11 17 AM |  |

## What areas of the 
site does it impact?
* Backend calls on 526 prefill
* display on maximum rating notices on rated disabilities page of 526

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/services/claim_fast_tracking/max_rating_annotator.rb [hunk 1]
```diff
@@ -18,11 +18,11 @@ def self.annotate_disabilities(rated_disabilities_response)
 
       ratings = get_ratings(diagnostic_codes)
       if ratings.present?
-        ratings.each do |rating|
-          rated_disability = rated_disabilities_response.rated_disabilities.find do |disability|
-            disability.diagnostic_code == rating['diagnostic_code']
+        rated_disabilities_response.rated_disabilities.each do |rated_disability|
+          max_rating = ratings.find do |rating|
+            rated_disability.diagnostic_code == rating['diagnostic_code']
```
Comment (nanotone) [inline app/services/claim_fast_tracking/max_rating_annotator.rb:23]: This isn't a new complaint about this region of code, but I feel like the nesting of the `find` inside the `each` makes for poor reading. (It also smells inefficient but the data structures are small). Do you want to try the route of building a DC➡️max_rating hash out of the VRO response, and see if that feels any better?

**Review by nanotone — APPROVED** [review]
💯

File: app/services/claim_fast_tracking/max_rating_annotator.rb (hunk 1 above)
Comment (SamStuckey) [reply app/services/claim_fast_tracking/max_rating_annotator.rb]: +1 that this is a bit of a strange pattern.  I would recommend putting in a helper method that describes what it is doing, or at the very least adding a comment that explains what the code is doing.

Comment (dfitchett) [reply app/services/claim_fast_tracking/max_rating_annotator.rb]: @SamStuckey Yeah it was. I cleaned it up in a more recent commit!

**Review by rmtolmach — APPROVED** [review]

**Review by mchae-nava — APPROVED** [review]
LGTM

**Pull request merged**